mitodl/ocw-data-parser
# Add several more fields to parsed JSON

#### What are the relevant tickets?
Closes #160
Closes #162
Closes #163 

#### What's this PR do?
- Adds `metadata_contributor_list`, `optional_text`, `optional_tab_title `, `resource_index_text`,  `highlights_text` and `related_content` to output JSON
- Reorders instructors based on their order in `metadata_contributor_list`

#### How should this be manually tested?
- Create an `example_courses.json` file containing the following:
```
{
    "courses": [
        "22-033-nuclear-systems-design-project-fall-2011",
        "18-06-linear-algebra-spring-2010",
        "20-219-becoming-the-next-bill-nye-writing-and-hosting-the-educational-show-january-iap-2015"
    ]
}
```
- In a python shell: run the following (after setting env variables for AWS):
```
from ocw_data_parser import OCWDownloader
downloader = OCWDownloader(
    "example_course.json", "PROD", "private/raw_courses", "ocw-content-storage"
)
downloader.download_courses()
parse_all(
     courses_dir="private/raw_courses",
     destination_dir="private/output,
     s3_bucket="mb_totally_fake",
     s3_links=False,
     upload_parsed_json=False,
     beautify_parsed_json=True,
     overwrite=True,
 )
 ```
- Check that the parsed JSON files in the output directories included `highlights_text` and `related_content` as keys and their values match what is in the `2.json` files from the `raw_courses` course folders.  Also, the resources should contain `optional_tab_title`, `optional_text`, and `resource_index_text`
- The output should also contain `metadata_contributor_list` and `instructors` fields.  Make sure that for the 20-219 site, the instructors are in the same order as shown in the screenshot for issue #160.
